Maple2.Server.Game/Manager/QuestManager.cs

```diff
@@ -0,0 +1,453 @@
+            session.Send(QuestPacket.Start(quest));
+        }
+
+    }
+
+    /// <summary>
+    /// Adds a quest to the quest manager.
+    /// </summary>
+    /// <param name="quest">Quest to add</param>
+    private void Add(Quest quest) {
+        if (quest.Metadata.Basic.Account > 0) {
+            accountValues.Add(quest.Id, quest);
+            return;
+        }
+        characterValues.Add(quest.Id, quest);
+    }
+
+    /// <summary>
+    /// Starts a new quest (or prexisting if repeatable).
+    /// </summary>
+    public void Start(int questId) {
+        if (characterValues.ContainsKey(questId) || accountValues.ContainsKey(questId)) {
+            // TODO: see if you can start the quest again
+            return;
+        }
+
+        if (!session.QuestMetadata.TryGet(questId, out QuestMetadata? metadata)) {
+            return;
+        }
+
+        if (!CanStart(metadata.Require)) {
+            return;
+        }
+
+        var quest = new Quest(metadata) {
+            Track = false,
+            State = QuestState.Started,
+            StartTime = DateTime.Now.ToEpochSeconds(),
+        };
+
+        for (int i = 0; i < metadata.Conditions.Length; i++) {
+            quest.Conditions.Add(i, new QuestCondition(metadata.Conditions[i]));
+        }
+
+        using GameStorage.Request db = session.GameStorage.Context();
+        quest = db.CreateQuest(session.CharacterId, quest);
+        if (quest == null) {
+            Console.WriteLine($"Quest ID was not created: {metadata.Id}");
+            return;
+        }
+        Add(quest);
+
+        // TODO: Confirm inventory can hold all the items.
+        foreach (QuestMetadataReward.Item acceptReward in metadata.AcceptReward.EssentialItem) {
+            Item? reward = session.Item.CreateItem(acceptReward.Id, acceptReward.Rarity, acceptReward.Amount);
+            if (reward == null) {
+                logger.Error("Failed to create quest reward {RewardId}", acceptReward.Id);
+                continue;
+            }
+            if (!session.Item.Inventory.Add(reward, true)) {
+                logger.Error("Failed to add quest reward {RewardId} to inventory", acceptReward.Id);
+            }
+        }
+
+        session.Send(QuestPacket.Start(quest));
+    }
+
+    /// <summary>
+    /// Updates all possible quests with the given condition type.
+    /// </summary>
+    /// <param name="type">Condition Type to update</param>
+    /// <param name="counter">Condition value to progress by. Default is 1.</param>
+    /// <param name="targetString">condition target parameter in string.</param>
+    /// <param name="targetLong">condition target parameter in long.</param>
+    /// <param name="codeString">condition code parameter in string.</param>
+    /// <param name="codeLong">condition code parameter in long.</param>
+    public void Update(ConditionType type, int counter = 1, string targetString = "", long targetLong = 0, string codeString = "", long codeLong = 0) {
```

Comment:
Is it possible to combine the logic with achievement conditions?

Reply:
Pushed an update where I combined the checks into one function but I think that might be as much as I can combine seeing as the rest is achievement/quest unique.